src/Model/ModelUtilities/TimeSelect.f90

```diff
@@ -1,19 +1,31 @@
-!> @brief Specify times for some event(s) to occur.
+!> @brief Specify times for some event to occur.
```

Comment:
Some of this code reminds me the STLVecInt dynamic vector.  We don't have a dynamic vector for dbl yet.  Just wanted to raise this similarity in case there were any reason to implement a dbl dynamic vector for this.

Reply:
Agreed it would be nice to use a more general data structure instead of hand-rolling it. I recall @mjr-deltares mentioned the possibility of templating vector types in FL earlier this year, maybe worth revisiting?

There has been discussion over on [stdlib](https://github.com/fortran-lang/stdlib) about containers including dynamic vector and linked list, but it seems like these won't arrive until generics make it into the language.

Relevant issues on stdlib are 68 and 598 (not linking to avoid cluttering them with an xref to mf6) and there is some work on generics [here](https://github.com/j3-fortran/generics)

Reply:
Yes, let's revisit @wpbonelli, I think this could be low hanging fruit.